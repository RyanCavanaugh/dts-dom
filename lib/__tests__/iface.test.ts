import {ParameterFlags, create, emit, type} from "../index"

describe("an interface type", () => {
    describe("with call signature", () => {
        it("matches the snapshot", () => {
            const parameters = [
                create.parameter('first', type.string, ParameterFlags.Optional),
                create.parameter('second', create.array('any'), ParameterFlags.Rest)
            ];
            const cs = create.callSignature(parameters, type.string);
            const iface = create.interface('iface');
          
            iface.members.push(cs);
          
            expect(emit(iface)).toMatchSnapshot();
        });
    });
});
