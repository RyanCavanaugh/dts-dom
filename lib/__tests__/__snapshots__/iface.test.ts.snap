// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`an interface type with call signature matches the snapshot 1`] = `
declare interface iface {
    (first?: string, ...second: any[]): string;
}


`;
